test/end2end_test.go

```diff
@@ -447,6 +448,8 @@ type test struct {
 	userAgent                   string
 	clientCompression           bool
 	serverCompression           bool
+	clientUseCompression        bool
```

Comment:
How is this different from clientCompression?  And clientNopCompression?  Please add a comment to all 4 of these fields.